### PR TITLE
used "tagged union" for union terminology

### PR DESCRIPTION
**Pull request by mccanne**
This commit is a cosmetic change that adopts the more conventional
terminology of tagged unions over the more obscure terms SplitZNG
and Selector.  No logic is changed; only names.

### CODE REVIEW

File: complex.go [hunk 1]
```diff
@@ -356,33 +356,33 @@ func (t *TypeUnion) ID() int {
...
 	return -1
 }
 
-// SplitZNG takes a zng encoding of a value of the receiver's union type and
-// returns the concrete type of the value, its selector, and the value encoding.
-// SplitZNG panics if the selector is invalid.
-func (t *TypeUnion) SplitZNG(zv zcode.Bytes) (Type, zcode.Bytes) {
-	if zv == nil {
+// Untag takes bytes of the reciever's type and returns the underlying value
+// as its types and bytes by removing the tag and determining that tag's
```
Comment (nwt) [inline complex.go:377]:
```suggestion
// as its type and bytes by removing the tag and determining that tag's
```

File: runtime/expr/shaper_test.go [hunk 2]
```diff
@@ -18,14 +18,14 @@ func TestBestUnionSelector(t *testing.T) {
 	u8named3, err := zctx.LookupTypeNamed("u8named3", u8)
 	require.NoError(t, err)
 
-	assert.Equal(t, -1, bestUnionSelector(u8, nil))
-	assert.Equal(t, -1, bestUnionSelector(u8, u8))
-	assert.Equal(t, -1, bestUnionSelector(zed.TypeUint16, zctx.LookupTypeUnion([]zed.Type{u8})))
+	assert.Equal(t, -1, bestUnionTag(u8, nil))
```
Comment (nwt) [inline runtime/expr/shaper_test.go:21]: Rename this test to TestBestUnionTag.

**Review by nwt — COMMENTED** [review]
zio/zng21io has a few references to selector that we should change.
```
$ git grep -i -n selector zio/zng21io
zio/zng21io/converter.go:100:           selector, err := zed21.DecodeInt(zcode.Bytes(zv))
zio/zng21io/converter.go:104:           inner, err := typ.Type(int(selector))
zio/zng21io/converter.go:110:           b.Append(zed.EncodeInt(int64(selector)))
zio/zng21io/zed21/type.go:8:    ErrUnionSelector = errors.New("union selector out of bounds")
zio/zng21io/zed21/union.go:16:// Type returns the type corresponding to selector.
zio/zng21io/zed21/union.go:17:func (t *TypeUnion) Type(selector int) (Type, error) {
zio/zng21io/zed21/union.go:18:  if selector < 0 || selector >= len(t.Types) {
zio/zng21io/zed21/union.go:19:          return nil, ErrUnionSelector
zio/zng21io/zed21/union.go:21:  return t.Types[selector], nil
```

**Comment by mccanne** [comment]
> zio/zng21io has a few references to selector that we should change.
> 
I know and wasn't going to bother.  That whole subdirectory is deprecated and 
will go away soon.

**Review by nwt — APPROVED** [review]

**Pull request merged**